cranelift/codegen/src/isa/aarch64/lower.isle

```diff
@@ -21,3 +21,128 @@
 
 (rule (lower (has_type ty (null)))
       (value_reg (imm ty 0)))
+
+;;;; Rules for `iadd` ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+
+;; `i64` and smaller
+
+;; base case, simply adding things in registers
+(rule (lower (has_type (fits_in_64 ty) (iadd x y)))
+      (value_reg (alu_rrr (iadd_op ty) (put_in_reg x) (put_in_reg y))))
+
+;; special case for when one operand is an immediate that fits in 12 bits
+(rule (lower (has_type (fits_in_64 ty) (iadd x (imm12_from_value y))))
```

Comment:
Not adding the commutative cases yet? I don't think we need to port only exactly the existing lowerings, and then add new lowerings at some indeterminate time in the future. I think it is fine to add new optimizations/rules as we port things to ISLE.